pandas/tests/frame/test_missing.py

```diff
@@ -671,6 +671,14 @@ def test_fillna_columns(self):
         expected = df.astype(float).fillna(method="ffill", axis=1)
         assert_frame_equal(result, expected)
 
+    def test_fillna_columns(self):
+        df = DataFrame(np.random.randn(10, 4))
```

Comment:
Can you add GH 17399 as a comment here?

Reply:
Also generally for test you can simplify by just constructing a DataFrame with literal values instead of random data - should make smaller and help readability

Reply:
I've added the comment (and properly fixed the test function name).  I'm afraid I found this bug on the first day of my learning to use pandas, so I'm not quite there yet to write a custom test.  Rather, I used/modified the existing one from test_fillna_columns()

Reply:
Hmm this test works on master for me without these changes - is this capturing the changed behavior?

Reply:
@jorvis can you respond here? Need to make sure we're actually fixing a real issue.

Reply:
I'll try to make a more direct test case. This is the demo that made me find it:

https://imgur.com/a/3ZrXqSv

After that I found an existing PR which had grown stale, and @jreback asked me to update it, so I thought I'd give it a try.

Reply:
I have updated it now to reflect a test similar to that Jupyter image above.